sdk/core/core/inc/az_span.h

```diff
@@ -135,161 +154,297 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
+ * also identical.
  */
-AZ_NODISCARD AZ_INLINE bool az_span_is_equal(az_span a, az_span b)
+AZ_NODISCARD AZ_INLINE bool az_span_is_content_equal(az_span span1, az_span span2)
 {
-  return az_span_length(a) == az_span_length(b)
-      && memcmp(az_span_ptr(a), az_span_ptr(b), az_span_length(a)) == 0;
+  return az_span_length(span1) == az_span_length(span2)
+      && memcmp(az_span_ptr(span1), az_span_ptr(span2), az_span_length(span1)) == 0;
 }
 
 /**
- * @brief Returns `true` if a content of the @a span is equal to a content of the @b
- * span using case-insensitive compare.
+ * @brief az_span_is_content_equal_ignoring_case returns `true` if the lengths and characters
+ * refered to by @span1 and @span2 are identical except for case. This function assumes the bytes in
+ * both spans are ASCII characters.
  *
- * @param a comparing span
- * @param b comparing span
- * @return AZ_NODISCARD az_span_is_equal_content_ignoring_case
+ * @return Returns true if the lengths of both spans are identical and the ASCII characters in both
+ * spans are also identical except for case.
  */
-AZ_NODISCARD bool az_span_is_content_equal_ignoring_case(az_span a, az_span b);
+AZ_NODISCARD bool az_span_is_content_equal_ignoring_case(az_span span1, az_span span2);
 
 /**
- * @brief convert all the span content to uint64
+ * @brief az_span_to_str copies a source span containing a string (not 0-terminated) to a
+ destination char buffer and appends the 0-terminating byte.
+
+ * The buffer refered to by destination must have a max_size that is at least 1 byte bigger
+ than the source az_span
+ *  the string in converts @b src span to zero-terminated str. Content is copied to @b
+ buffer and then \0 is
+ * addeed at the end. Then out_result will be created out of buffer
  *
- * @param self source span
- * @param out result span
- * @return AZ_NODISCARD az_span_to_uint64
+ * @param[in] destination A pointer to a buffer where the string should be copied
+ * @param[in] destination_max_size The maximum size of the buffer refered to by destination
+ * @param[in] source The az_span containing the not-0-terminated string
+ * @return #AZ_OK if the span's string is copied sucessfully to the destination.
+ *         #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the destination buffer is too small to
+ copy the string and 0-terminate it
  */
-AZ_NODISCARD az_result az_span_to_uint64(az_span self, uint64_t* out);
+AZ_NODISCARD az_result
+az_span_to_str(char* destination, int32_t destination_max_size, az_span source);
+
+/******************************  SPAN PARSING */
 
 /**
- * @brief convert all the span content to uint32
+ * @brief az_span_to_uint64 parses an az_span containing ASCII digits into a uint64 number
  *
- * @param self source span
- * @param out result span
- * @return AZ_NODISCARD az_span_to_uint32
+ * @param[in] span The az_span containing the ASCII digits to be parsed.
+ * @param[in] out The pointer to the variable that is to receive the number
```

Comment:
CI failing because of docs mismatch. Should be `out_number`.
```text
/Users/runner/runners/2.165.1/work/1/s/sdk/core/core/inc/az_span.h:231:15: note: did you mean 'out_number'?
 * @param[in] out The pointer to the variable that is to receive the number
```